File: CompilerRR/CompilerRR/src/ch/fhnw/cpib/compiler/classes/RepeatingOptionalProgramParameters.java
package ch.fhnw.cpib.compiler.classes;

import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn;
import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn.IOptionalCHANGEMODE;
import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn.IOptionalFLOWMODE;
import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn.IRepeatingOptionalProgramParameters;
import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn.ITypedIdent;
import ch.fhnw.cpib.compiler.scanner.Token;

public class RepeatingOptionalProgramParameters implements IConcSyn.IRepeatingOptionalProgramParameters{

	Token comma;
	IConcSyn.IOptionalFLOWMODE flowMode;
	IConcSyn.IOptionalCHANGEMODE changeMode;
	IConcSyn.ITypedIdent typedIdent;
	IConcSyn.IRepeatingOptionalProgramParameters repOptProPar;
	public RepeatingOptionalProgramParameters(Token comma, IOptionalFLOWMODE flowMode, IOptionalCHANGEMODE changeMode,
			ITypedIdent typedIdent, IRepeatingOptionalProgramParameters repOptProPar) {
		super();
		this.comma = comma;
		this.flowMode = flowMode;
		this.changeMode = changeMode;
		this.typedIdent = typedIdent;
		this.repOptProPar = repOptProPar;
	}
	
	
	
}

File: CompilerMR/IML_Compiler/IMLParser/DataTypes/Concrete/RepeatingOptionalProgramParameters.cs
using IML_Compiler.IMLParser.DataTypes.Abstract;

namespace IML_Compiler.IMLParser.DataTypes.Concrete
{
    class RepeatingOptionalProgramParametersEps : IRepeatingOptionalProgramParameters
    {
        public IAbsList<AbsProgramParameter> ToAbySyn()
        {
            return new AbsEmptyList<AbsProgramParameter>();
        }
    }

    class RepeatingOptionalProgramParametersParams : IRepeatingOptionalProgramParameters
    {
        public IOptionalFlowMode FlowMode { get; private set; }
        public IOptionalChangeMode ChangeMode { get; private set; }
        public ITypedIdent TypedIdent { get; private set; }
        public IRepeatingOptionalProgramParameters RepeatingOptionalProgramParameters { get; private set; }

        public RepeatingOptionalProgramParametersParams(IOptionalFlowMode fmode, IOptionalChangeMode cmode,
            ITypedIdent tident, IRepeatingOptionalProgramParameters oparams)
        {
            FlowMode = fmode;
            ChangeMode = cmode;
            TypedIdent = tident;
            RepeatingOptionalProgramParameters = oparams;
        }

        public IAbsList<AbsProgramParameter> ToAbySyn()
        {
            var abs = TypedIdent.ToAbySyn();
            return new AbsProgramParameterList()
            {
                ProgramParameter =
                    new AbsProgramParameter()
                    {
                        ChangeMode = ChangeMode.ToAbySyn(),
                        FlowMode = FlowMode.ToAbsSyn(),
                        Ident = abs.Ident,
                        Type = abs.Type
                    },
                ProgramParameterList = RepeatingOptionalProgramParameters.ToAbySyn()
            };
        }
    }
}
